Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrTextEmitter.kt

```diff
@@ -0,0 +1,657 @@
+            "@ExternalObjCClass" + when {
+                annotationStub.protocolGetter.isEmpty() && annotationStub.binaryName.isEmpty() -> ""
+                annotationStub.protocolGetter.isEmpty() -> "($binaryName)"
+                annotationStub.binaryName.isEmpty() -> "($protocolGetter)"
+                else -> "($protocolGetter, $binaryName)"
+            }
+        }
+        AnnotationStub.CCall.CString ->
+            "@CCall.CString"
+        AnnotationStub.CCall.WCString ->
+            "@CCall.WCString"
+        is AnnotationStub.CCall.Symbol ->
+            "@CCall(${annotationStub.symbolName.quoteAsKotlinLiteral()})"
+        is AnnotationStub.CStruct ->
+            "@CStruct(${annotationStub.struct.quoteAsKotlinLiteral()})"
+        is AnnotationStub.CNaturalStruct ->
+            "@CNaturalStruct(${annotationStub.struct.quoteAsKotlinLiteral()})"
+        is AnnotationStub.CLength ->
+            "@CLength(${annotationStub.length})"
+        is AnnotationStub.Deprecated ->
+            "@Deprecated(${annotationStub.message.quoteAsKotlinLiteral()}, " +
+                    "ReplaceWith(${annotationStub.replaceWith.quoteAsKotlinLiteral()}), " +
+                    "DeprecationLevel.ERROR)"
+    }
+
+    private fun renderEnumEntry(enumEntryStub: EnumEntryStub): String =
+            "${enumEntryStub.name.asSimpleName()}(${renderValueUsage(enumEntryStub.constant)})"
+
+    private fun renderGetter(accessor: PropertyAccessor.Getter): String {
+        val annotations = accessor.annotations.joinToString(separator = "") { renderAnnotation(it) + " " }
+
+        return annotations + if (accessor is PropertyAccessor.Getter.ExternalGetter) {
+            "external get"
+        } else {
+            "get() = ${renderPropertyAccessorBody(accessor)}"
+        }
+    }
+
+    private fun renderSetter(accessor: PropertyAccessor.Setter): String {
+        val annotations = accessor.annotations.joinToString(separator = "") { renderAnnotation(it) + " " }
+        return annotations + if (accessor is PropertyAccessor.Setter.ExternalSetter) {
+            "external set"
+        } else {
+            "set(value) { ${renderPropertyAccessorBody(accessor)} }"
+        }
+    }
+
+    private fun renderPropertyAccessorBody(accessor: PropertyAccessor): String = when (accessor) {
+        is PropertyAccessor.Getter.SimpleGetter -> {
+            when {
+                accessor in propertyAccessorBridgeBodies -> propertyAccessorBridgeBodies.getValue(accessor)
+                accessor.constant != null -> renderValueUsage(accessor.constant)
+                else -> error("Bridge body for getter was not generated")
+            }
+        }
+
+        is PropertyAccessor.Getter.ArrayMemberAt -> "arrayMemberAt(${accessor.offset})"
+
+        is PropertyAccessor.Getter.MemberAt -> {
+            if (accessor.typeArguments.isEmpty()) {
+                "memberAt(${accessor.offset})"
```

Comment:
The presence of `.value` shouldn't be deduced from emptiness of type arguments.
Additional flag better be added for this.